### PR TITLE
Fixes #77 Including SWIG interface files

### LINKED ISSUE #77: Python-specific SWIG files
- [ ] Move to this repo `python_*` SWIG header files added in skycoin/skycoin#1641 .
- [ ] Create a new header file for Python-related definitions included in generic interface files .

### PR DESCRIPTION
**Pull request by stdevHan**
Fixes #77 

Changes:
-Transferring specific SWIG interface files

Does this change need to mentioned in CHANGELOG.md?

Yes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@ SKYCOIN_DIR = gopath/src/github.com/skycoin/skycoin
 SKYBUILD_DIR = $(SKYCOIN_DIR)/build
 BUILDLIBC_DIR = $(SKYBUILD_DIR)/libskycoin
 LIBC_DIR = $(SKYCOIN_DIR)/lib/cgo
-LIBSWIG_DIR = $(SKYCOIN_DIR)/lib/swig
+LIBSWIG_DIR = swig
```
Comment (olemis) [inline Makefile]: revert , keep it as it was before

File: swig/include/swig.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (olemis) [inline swig/include/swig.h]: No . Retrieve this from submodule .

File: swig/python_pubkeys.i [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+		int res = SWIG_ConvertPtr(o, &argp, SWIGTYPE_p_cipher_PubKey, 0 | 0);
+		if (!SWIG_IsOK(res))
+			SWIG_exception_fail(SWIG_TypeError, "expecting type PubKey");
+		cipher_PubKey* p = (cipher_PubKey*)argp;
+		memcpy(pdata, p, sizeof(cipher_PubKey));
+		pdata++;
+	}
+}
+
+%typemap(freearg) (cipher_PubKeys* __in_pubKeys) {
```
Comment (olemis) [inline swig/python_pubkeys.i]: I do not think this type exists anymore , does it ?

Comment (olemis) [reply swig/python_pubkeys.i]:
```
$ grep -nr cipher_PubKeys src/
```

no results

File: swig/python_pubkeys.i [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/*cipher_PubKeys* input parameter */
```
Comment (olemis) [inline swig/python_pubkeys.i]: @stdevHan shall we remove this ?

Comment (olemis) [inline swig/python_pubkeys.i]: @stdevHan shall we remove this ?

File: swig/skycoin.i [hunk 5]
```diff
@@ -0,0 +1,26 @@
+%module skycoin
```
Comment (olemis) [inline swig/skycoin.i]: Rename `skycoin.i` => `pyskycoin.i`

File: swig/structs_typemaps.i [hunk 6]
```diff
@@ -0,0 +1,62 @@
+/*
```
Comment (olemis) [inline swig/structs_typemaps.i]: Is this specific to Python ?

File: swig/skycoin.mem.i [hunk 7]
```diff
@@ -0,0 +1,443 @@
+/**
```
Comment (olemis) [inline swig/skycoin.mem.i]: This file is generic . Move to `lib/swig/common` in `skycoin/skycoin` .

File: swig/skycoin.mem.i [hunk 8]
```diff
@@ -0,0 +1,443 @@
...
+
+%typecheck(SWIG_TYPECHECK_INTEGER) Transactions__Handle {
+  $1 = PyInt_Check($input) ? 1 : 0;
+}
+
+%typecheck(SWIG_TYPECHECK_INTEGER) AddressUxOuts_Handle {
+  $1 = PyInt_Check($input) ? 1 : 0;
+}
+
+#if defined(SWIGPYTHON)
```
Comment (olemis) [inline swig/skycoin.mem.i]: only this part looks like specific to Python , so keep it here , please

File: swig/skycoin.mem.i [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+
+%apply Handle* { Wallet__Handle*, Options__Handle*, ReadableEntry__Handle*, ReadableWallet__Handle*, WebRpcClient__Handle*,
+	WalletResponse__Handle*, Client__Handle*, Strings__Handle*, Wallets__Handle*, Config__Handle*,
+	App__Handle*, Context__Handle*, GoStringMap_*, PasswordReader__Handle*,
+	Transaction__Handle*, Transactions__Handle*, CreatedTransaction__Handle*,
+	CreatedTransactionOutput__Handle*, CreatedTransactionInput__Handle*, CreateTransactionResponse__Handle*,
+	Block__Handle*, SignedBlock__Handle*, BlockBody__Handle*, BuildInfo_Handle*, Number_Handle*, Signature_Handle*,AddressUxOuts_Handle*
+	}
+
+%typecheck(SWIG_TYPECHECK_INTEGER) Transaction__Handle {
```
Comment (olemis) [inline swig/skycoin.mem.i]: These are specific to Python . Keep them in this repo as well .

File: swig/skycoin.mem.i [hunk 10]
```diff
@@ -0,0 +1,443 @@
...
+		GoUint32 result = SKY_cipher_GenerateDeterministicKeyPairsSeed(seed, n, newSeed, &data);
+		if( result == 0){
+			__out_secKeys->data = data.data;
+			__out_secKeys->count = data.len;
+		}
+		return result;
+	}
+}
+
+// %rename(SKY_cipher_PubKeySlice_Len) wrap_SKY_cipher_PubKeySlice_Len;
```
Comment (olemis) [inline swig/skycoin.mem.i]: Remove

File: swig/skycoin.mem.i [hunk 11]
```diff
@@ -0,0 +1,443 @@
...
+// 		GoSlice_ data;
+// 		data.data = __in_pubKeys->data;
+// 		data.len = __in_pubKeys->count;
+// 		data.cap = __in_pubKeys->count;
+// 		GoUint32 result = SKY_cipher_PubKeySlice_Len(&data);
+// 		return result;
+// 	}
+// }
+
+// %rename(SKY_cipher_PubKeySlice_Less) wrap_SKY_cipher_PubKeySlice_Less;
```
Comment (olemis) [inline swig/skycoin.mem.i]: Remove

File: swig/skycoin.mem.i [hunk 12]
```diff
@@ -0,0 +1,443 @@
...
+// 		GoSlice_ data;
+// 		data.data = __in_pubKeys->data;
+// 		data.len = __in_pubKeys->count;
+// 		data.cap = __in_pubKeys->count;
+// 		GoUint32 result = SKY_cipher_PubKeySlice_Less(&data, p1, p2);
+// 		return result;
+// 	}
+// }
+
+// %rename(SKY_cipher_PubKeySlice_Swap) wrap_SKY_cipher_PubKeySlice_Swap;
```
Comment (olemis) [inline swig/skycoin.mem.i]: Remove

File: swig/structs.i [hunk 13]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (olemis) [inline swig/structs.i]: isn't this generic ? if so, pls, move to `skycoin/skycoin` .

File: swig/structs_typemaps.i (hunk 6 above)
Comment (olemis) [reply swig/structs_typemaps.i]: but is it specific to Python or not ? I see no reference to any Python-specific feature .

Comment (olemis) [reply swig/structs_typemaps.i]: this whole file looks generic to me

File: swig/structs.i (hunk 13 above)
Comment (stdevHan) [reply swig/structs.i]: https://github.com/skycoin/pyskycoin/blob/develop/Makefile#L44

**Pull request merged**